### PR TITLE
Feature/ees 1335 add methodology approval status apis

### PR DESCRIPTION
**Pull request by duncan-at-hiveit**
This PR needs to go in after EES-1308.

This PR:
- adds API endpoints and security for the Methodology approval process
- cleans up security config by extracting from Startup.cs and organising Policies / Claims / Auth Handlers by logical groupings

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/20200218140122_AddMissingGeoJsonFlagToDataBlockRequests.Designer.cs [hunk 1]
```diff
@@ -0,0 +1,3862 @@
+﻿// <auto-generated />
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/20200218140122_AddMissingGeoJsonFlagToDataBlockRequests.Designer.cs]: This change has already gone in, see 20200214160903_UpdateDataBlockFields.cs
I see this is doing a lot more than just adding the missing geojson flag too.

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/20200218140122_AddMissingGeoJsonFlagToDataBlockRequests.Designer.cs [hunk 2]
```diff
@@ -0,0 +1,3862 @@
...
...nd 15 who are not boarders as of the January school census each year.\\n      </td>\\n    </tr>\\n  </tfoot>\\n</table>\\n<p>\\n  In published absence statistics, pupil enrolments who first enrolled at a\\n  school within the second half of the summer term are not included.\\n</p>\\n<p>\\n  This is to ensure the same cohorts of enrolments are included in both the five\\n  and six half term absence measures.\\n</p>\\n<h3 id=\\\"section1-7\\\">\\n  1.7 Published geographical and characteristics breakdowns\\n</h3>\\n<p>\\n  DfE routinely publishes pupil absence information at national, local authority\\n  and school level - including breakdowns by pupil characteristics.\\n</p>\\n<p>\\n  The autumn term absence publication provides high level information designed\\n  to give an early indication on absence levels and the effect of winter\\n  illness. This includes:\\n</p>\\n<ul class=\\\"govuk-list govuk-list--bullet\\\">\\n  <li>authorised, unauthorised and overall absence rates</li>\\n  <li>absence broken down by reason</li>\\n  <li>\\n    the number of pupils with one or more sessions of absence for different\\n    reasons\\n  </li>\\n  <li>information on persistent absence</li>\\n</
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/20200218140122_AddMissingGeoJsonFlagToDataBlockRequests.Designer.cs]: Order has been removed from Release now

Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/20200218140122_AddMissingGeoJsonFlagToDataBlockRequests.Designer.cs]: Should these changes be in here?

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs [hunk 3]
```diff
@@ -139,7 +139,7 @@ public static IQueryable<Publication> HydratePublicationForPublicationViewModel(
                 }
 
                 // TODO: this will want updating when methodology status is fully implemented
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs]: This TODO can be removed now

File: tests/newman/files/admin-api-test.csv [hunk 4]
```diff
@@ -1,161 +1,16165 @@
...
-2020,Calendar year,Ward,E92000001,England,,,,,,,,,,,,,,,E05010450,Rural West York,6650
-2012,Calendar year,Ward,E92000001,England,,,,,,,,,,,,,,,E05000364,Syon,1109
-2018,Calendar year,Ward,E92000001,England,,,,,,,,,,,,,,,E05006937,Yoxall,1926
-2012,Calendar year,Ward,E92000001,England,,,,,,,,,,,,,,,E05010291,Nailsea Youngwood,8150
-2016,Calendar year,Ward,E92000001,England,,,,,,,,,,,,,,,E05010450,Rural West York,4035
-2008,Calendar year,Ward,E92000001,England,,,,,,,,,,,,,,,E05000364,Syon,5505
-2007,Calendar year,Ward,E92000001,England,,,,,,,,,,,,,,,E05006937,Yoxall,7499
-2011,Calendar year,Ward,E92000001,England,,,,,,,,,,,,,,,E05010291,Nailsea Youngwood,9603
-2007,Calendar year,Ward,E92000001,England,,,,,,,,,,,,,,,E05010450,Rural West York,9884
+{
```
Comment (benoutram) [inline tests/newman/files/admin-api-test.csv]: This doesn't look right!

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs (hunk 3 above)
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationService.cs]: Actually was deliberately keeping this TODO in place as a reminder that we still need to get the publisher to update the Methodology status and create a "Live" methodology status value as well as approved.

File: tests/newman/files/admin-api-test.csv (hunk 4 above)
Comment (duncan-at-hiveit) [reply tests/newman/files/admin-api-test.csv]: haha :)  No it does not - thanks for the assist!!

Comment (duncan-at-hiveit) [reply tests/newman/files/admin-api-test.csv]: Looks like @markhiveit fixed up already - my thanks!

**Pull request merged**